rfc/2015-06-10-message-api.markdown

```diff
@@ -0,0 +1,69 @@
+---
+title: API for messages
+category: rfc
+status: proposal
+---
+
+This RFC describe the CaliOpen message API requirements to implement current
+mockups.
+
+## Resources
+
+> NOTE the field names and structure are here to describe pieces of information.
+> Feel free to provide better ones.
+
+> NOTE the term **Link** is used for nested resources. The implementation can
+> embed the information OR provide information to retrieve it later (id,
+> href...).
+>
+> If the information is not embedded, then there should be an API endpoint to
+> retrieve missing information in batch and prevent too many requests.
+
+Message related resources are **Thread** and **Message**.
+
+A **Thread** is a group of **Messages** with additional pieces of information.
+Every **Thread** MUST provide the following information:
+
+* `total_count`: the number of messages
+* `unread_count`: the number of unread messages
+* `subject`: the thread subject. Empty string if none.
+* `contacts`: a list of links to contacts involved in the conversation
+* `importance_level`: the importance level associated to message
+* `privacy_index`: the privacy index associated to message
+* `messages`: a list of links to most recent messages. **<--- This is used for
```

Comment:
Thread.messages is not possible without important impact on performance if API give it this way.
There is a /threads/<thread_id>/messages route to use for that